File: src/main/java/com/flexpoker/game/query/repository/OpenGameForPlayerRepository.java
package com.flexpoker.game.query.repository;

import java.util.List;
import java.util.UUID;

import com.flexpoker.game.query.dto.GameStage;
import com.flexpoker.game.query.dto.OpenGameForUser;

public interface OpenGameForPlayerRepository {

    List<OpenGameForUser> fetchAllOpenGamesForPlayer(UUID playerId);

    void deleteOpenGameForPlayer(UUID playerId, UUID gameId);

    void addOpenGameForUser(UUID playerId, UUID gameId, String gameName);

    void changeGameStage(UUID playerId, UUID gameId, GameStage gameStage);

    void assignTableToOpenGame(UUID playerId, UUID gameId, UUID tableId);

}

File: src/main/kotlin/com/flexpoker/game/query/repository/OpenGameForPlayerRepository.kt
package com.flexpoker.game.query.repository

import com.flexpoker.game.command.GameStage
import com.flexpoker.game.query.dto.OpenGameForUser
import java.util.UUID

interface OpenGameForPlayerRepository {
    fun fetchAllOpenGamesForPlayer(playerId: UUID): List<OpenGameForUser>?
    fun deleteOpenGameForPlayer(playerId: UUID, gameId: UUID)
    fun addOpenGameForUser(playerId: UUID, gameId: UUID, gameName: String)
    fun changeGameStage(playerId: UUID, gameId: UUID, gameStage: GameStage)
    fun assignTableToOpenGame(playerId: UUID, gameId: UUID, tableId: UUID)
}